filters/include/pcl/filters/spherical_voxel_grid.h

```diff
@@ -0,0 +1,357 @@
+          return;
+        }
+
+        leaf_size_r_ = r;
+        leaf_size_theta_ = static_cast<float>(M_PI) /  vertical_divisions;
+        leaf_size_phi_ = (2.0f * static_cast<float>(M_PI)) / horizontal_divisions;
+
+        leaf_theta_divisions_ = vertical_divisions;
+        leaf_phi_divisions_ = horizontal_divisions;
+      }
+
+      /** \brief Get the radial size of the leaves
+        */
+      inline float
+      getLeafSizeR () { return (leaf_size_r_); }
+
+      /** \brief Get the vertical angular size (radians) of the leaves
+        */
+      inline float
+      getLeafSizeVertical () { return (leaf_size_theta_); }
+
+      /** \brief Get the horizontal angular size (radians) of the leaves
+        */
+      inline float
+      getLeafSizeHorizontal () { return (leaf_size_phi_); }
+
+      /** \brief Get the number of radial divisions
+        */
+      inline int
+      getDivisionsR () { return (leaf_r_divisions_); }
+
+      /** \brief Get the number of vertical angular divisions
+        */
+      inline int
+      getDivisionsVertical () { return (leaf_theta_divisions_); }
+
+      /** \brief Get the number of horizontal angular divisions
+        */
+      inline int
+      getDivisionsHorizontal () { return (leaf_phi_divisions_); }
+
+      /** \brief Set to true if all fields need to be downsampled, or false if just XYZ.
+        * \param[in] downsample the new value (true/false)
+        */
+      inline void
+      setDownsampleAllData (bool downsample) { downsample_all_data_ = downsample; }
+
+      /** \brief Get the state of the internal downsampling parameter (true if
+        * all fields need to be downsampled, false if just XYZ).
+        */
+      inline bool
+      getDownsampleAllData () { return (downsample_all_data_); }
+
+      /** \brief Set the minimum number of points required for a voxel to be used.
+        * \param[in] min_points_per_voxel the minimum number of points for required for a voxel to be used
+        */
+      inline void
+      setMinimumPointsNumberPerVoxel (unsigned int min_points_per_voxel) { min_points_per_voxel_ = min_points_per_voxel; }
+
+      /** \brief Return the minimum number of points required for a voxel to be used.
+        */
+      inline unsigned int
+      getMinimumPointsNumberPerVoxel () { return (min_points_per_voxel_); }
+
+      /** \brief Return the maximum radius of a point from the ceter,
+        * i.e. the radius of the spherical voxel grid (after filtering)
+        */
+      inline float
+      getMaximumRadius () { return (max_radius_); }
+
+      /** \brief Set the cartesian origin used to build the spherical grid.
+        * \param[in] origin the cartesian origin
+        */
+      inline void
+      setOrigin (const Eigen::Vector4f& origin) { filter_origin_ = origin; }
+
+      /** \brief Get the cartesian origin used to build the spherical grid.
+        */
+      inline Eigen::Vector3f
```

Comment:
Is there any pragmatic reason to have 3-vector in `getOrigin` but 4-vector in `setOrigin`?

Reply:
No. I believe this was mimicking some of the behavior in `VoxelGrid`. Is there a particular reason to prefer 4 vs 3 vectors (SIMD instructions or something)? If not, I'll switch `setOrigin` to take a 3-vector since it needs 3-dimensional coordinates.

Reply:
Taking into account how this is used in the class (single vector subtraction inside a fat loop), I would be very surprised to see any performance difference. I'd back switching to 3-vectors.